### ISSUE TITLE
[JENKINS-71496] WCAG Operable:2.1 - Keyboard operation and navigation - Result 4. Important content cannot be accessed and operated using the keyboard

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#7341]
<p>Individual important content is only visualized as a tool tip when operated with the mouse. Keyboard users and screen reader users therefore have no way of having this information displayed or read out. The following examples stood out:</p>
<ul>
	<li>"Weather icon in Branches table: a tooltip above the icon shows a short report summarizing the results of the latest builds</li>
	<li>Success message and log file in Stage View table: The success message "Success" and the option to open and display the log file are only displayed as a tooltip when using the mouse.
	<ul>
		<li>This means that blind users and users with motor disabilities are denied the opportunity to take a detailed look at the log file into the build process.</li>
		<li>This is considered serious, as the usability test in particular has shown that the log file is an important tool for users.</li>
		<li>After opening the log button, the log file is displayed.</li>
		<li>Note: The log file is not in the correct tab order. Even if opening with the keyboard were possible, the focus must also be placed in the opened log file.</li>
	</ul>
	</li>
	<li>Legends for the trend charts: In the branch view, the various charts provide detailed information about the course, concrete values of the various graphs at specific times, and display pop-ups that can only be made visible when using the mouse.
	<ul>
		<li>Note: There is an alternative tabular presentation of the values for individual evaluations, provided that the corresponding menu items are selected and displayed, e.g. B. for "CheckStyle Warnings". For the representations of “Trend of the test results" and "Aggregated Analysis Results" this is not the case</li>
	</ul>
	</li>
	<li>Context menus offer additional options in various places, but can only be operated with the mouse</li>
	<li>At one point on the branch page, the menu selectors are hidden even though they are there, as hovering brings to view</li>
	<li>In the Stage View, the list of commits can only be visualized as a tool tip using the mouse:
	<ul>
		<li>The operation of an application should be enabled device-independent.</li>
		<li>This means that all UI elements must be operable with both the mouse and the keyboard.</li>
		<li>If individual elements only have a mouseover event, they cannot be operated using the keyboard.</li>
		<li>All UI elements that can be operated with the mouse must also be accessible with the keyboard.</li>
		<li>Corresponding events for keyboard operation may also have to be added for this purpose. In addition, the focus management must be adjusted so that the focus order corresponds to the order of operation.</li>
	</ul>
	</li>
</ul>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cristinap">cristinap</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-71496" target="_blank">WCAG Operable:2.1 - Keyboard operation and navigation - Result 4. Important content cannot be accessed and operated using the keyboard</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>label(s)</b>: accessibility
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2023-07-26T15:39:09+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Individual important content is only visualized as a tool tip when operated with the mouse. Keyboard users and screen reader users therefore have no way of having this information displayed or read out. The following examples stood out:</p>
<ul>
	<li>"Weather icon in Branches table: a tooltip above the icon shows a short report summarizing the results of the latest builds</li>
	<li>Success message and log file in Stage View table: The success message "Success" and the option to open and display the log file are only displayed as a tooltip when using the mouse.
	<ul>
		<li>This means that blind users and users with motor disabilities are denied the opportunity to take a detailed look at the log file into the build process.</li>
		<li>This is considered serious, as the usability test in particular has shown that the log file is an important tool for users.</li>
		<li>After opening the log button, the log file is displayed.</li>
		<li>Note: The log file is not in the correct tab order. Even if opening with the keyboard were possible, the focus must also be placed in the opened log file.</li>
	</ul>
	</li>
	<li>Legends for the trend charts: In the branch view, the various charts provide detailed information about the course, concrete values of the various graphs at specific times, and display pop-ups that can only be made visible when using the mouse.
	<ul>
		<li>Note: There is an alternative tabular presentation of the values for individual evaluations, provided that the corresponding menu items are selected and displayed, e.g. B. for "CheckStyle Warnings". For the representations of “Trend of the test results" and "Aggregated Analysis Results" this is not the case</li>
	</ul>
	</li>
	<li>Context menus offer additional options in various places, but can only be operated with the mouse</li>
	<li>At one point on the branch page, the menu selectors are hidden even though they are there, as hovering brings to view</li>
	<li>In the Stage View, the list of commits can only be visualized as a tool tip using the mouse:
	<ul>
		<li>The operation of an application should be enabled device-independent.</li>
		<li>This means that all UI elements must be operable with both the mouse and the keyboard.</li>
		<li>If individual elements only have a mouseover event, they cannot be operated using the keyboard.</li>
		<li>All UI elements that can be operated with the mouse must also be accessible with the keyboard.</li>
		<li>Corresponding events for keyboard operation may also have to be added for this purpose. In addition, the focus management must be adjusted so that the focus order corresponds to the order of operation.</li>
	</ul>
	</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<i>[Epic: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-71153%20label%3Aepic">JENKINS-71153</a>]</i>
